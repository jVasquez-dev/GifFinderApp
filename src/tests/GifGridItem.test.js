import { shallow } from 'enzyme';
import { GifGridItem } from '../GifGridItem';
import React from 'react';

describe('pruebas GifGridItem', () => {

    const title = 'My title';
    const url = 'https://localhost/imagen.jpg';

    const wrapper = shallow(<GifGridItem title={ title } url={ url }/>);
    
    test('Debe de mostrar <GifGridItem /> correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    })
    
    test('should have a paragraph with the value of title', () => {
       
        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe( title )
    })
    
    test('should have an img tag with url and alt as properties', () => {
        
        const img = wrapper.find('img');
        expect( img.prop('src') ).toBe( url );
        expect( img.prop('alt') ).toBe( title );

    })

    test('should have a div with the class "animate__zoomIn"', () => {
        
        const div = wrapper.find('.card');
        expect( div.prop('className').includes('animate__zoomIn') ).toBe( true );
    })
    
    
})