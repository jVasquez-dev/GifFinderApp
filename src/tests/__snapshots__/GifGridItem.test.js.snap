// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas GifGridItem Debe de mostrar <GifGridItem /> correctamente 1`] = `
<div
  className="card animate__animated animate__zoomIn"
>
  <img
    alt="My title"
    src="https://localhost/imagen.jpg"
  />
  <p>
    My title
  </p>
</div>
`;
